kamu-core/src/domain/sync_service.rs

```diff
@@ -164,12 +166,24 @@ impl DatasetNotFoundError {
 ///////////////////////////////////////////////////////////////////////////////
 
 #[derive(Error, Clone, Eq, PartialEq, Debug)]
-#[error("Source and destination datasets have diverged, source head is {src_head}, destination head is {dst_head}")]
+#[error("Source and destination datasets have diverged, source head is '{src_head}', ahead by {uncommon_blocks_in_src} blocks, destination head is {dst_head}, ahead by {uncommon_blocks_in_dst}")]
```

Comment:
How about we borrow the phrasing of this error from git:
```rust
"Source and destination datasets have diverged and have {uncommon_blocks_in_src} and {uncommon_blocks_in_dst} different blocks respectively, source head is {src_head}, destination head is {dst_head}"
```

Reply:
Looks good